instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/_server.py

```diff
@@ -56,68 +83,97 @@ def cancel(self, *args, **kwargs):
     def add_callback(self, *args, **kwargs):
         return self._servicer_context.add_callback(*args, **kwargs)
 
+    def disable_next_message_compression(self):
+        return self._service_context.disable_next_message_compression()
+
     def invocation_metadata(self, *args, **kwargs):
         return self._servicer_context.invocation_metadata(*args, **kwargs)
 
-    def peer(self, *args, **kwargs):
-        return self._servicer_context.peer(*args, **kwargs)
+    def peer(self):
+        return self._servicer_context.peer()
+
+    def peer_identities(self):
+        return self._servicer_context.peer_identities()
 
-    def peer_identities(self, *args, **kwargs):
-        return self._servicer_context.peer_identities(*args, **kwargs)
+    def peer_identity_key(self):
+        return self._servicer_context.peer_identity_key()
 
-    def peer_identity_key(self, *args, **kwargs):
-        return self._servicer_context.peer_identity_key(*args, **kwargs)
+    def auth_context(self):
+        return self._servicer_context.auth_context()
 
-    def auth_context(self, *args, **kwargs):
-        return self._servicer_context.auth_context(*args, **kwargs)
+    def set_compression(self, compression):
+        return self._servicer_context.set_compression(compression)
 
     def send_initial_metadata(self, *args, **kwargs):
         return self._servicer_context.send_initial_metadata(*args, **kwargs)
 
     def set_trailing_metadata(self, *args, **kwargs):
         return self._servicer_context.set_trailing_metadata(*args, **kwargs)
 
-    def abort(self, *args, **kwargs):
-        if not hasattr(self._servicer_context, "abort"):
-            raise RuntimeError(
-                "abort() is not supported with the installed version of grpcio"
+    def abort(self, code, details):
+        self._active_span.set_attribute("grpc.status_code", code.name)
+        self._active_span.set_attribute("grpc.details", details)
```

Comment:
Let me do some testing with this, but I think you're probably right.

Reply:
Okay maybe I've got something else wrong, but if I don't set these attributes, they don't show up in my tracing report (using Zipkin) - is `set_status()` supposed to set them?  I looked at a few other instrumentation libraries, and they seem to set similar attributes:
* https://github.com/open-telemetry/opentelemetry-python/blob/master/instrumentation/opentelemetry-instrumentation-falcon/src/opentelemetry/instrumentation/falcon/__init__.py#L216
* https://github.com/open-telemetry/opentelemetry-python/blob/master/instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py#L154

I will make them `rpc` instead of `grpc` though.

Reply:
It should be implemented in the Zipkin exporter to support that. I see this code in there https://github.com/open-telemetry/opentelemetry-python/blob/82356a8ee94b01003c118944dcbe9923cd35e8d7/exporter/opentelemetry-exporter-zipkin/src/opentelemetry/exporter/zipkin/__init__.py#L194-L201

Does that do what you expect on the Zipkin side?

Reply:
I'm using the current `ZipkinSpanExporter` from the same commit here, and I don't see any `otel` attributes set when viewing a span in the UI, nor is there any other way to tell the trace is considered an error.  I feel like I'm missing something here - I'll keep digging.

I'm a bit new to using all of this, note :)

Reply:
No worries 😄 I'm not too familiar with zipkin either, but my understanding is that exporters can represent `span.status` in their own format, whatever that may be. I think this would be a separate issue to this instrumentation PR

Reply:
I'll try it with Jaeger, I did some initial testing with that as well.  I do suspect it's not this PR, but just want to be sure.

Reply:
@aabmass You're right!  I set up a test tonight using Jaeger instead of Zipkin and an error trace is indeed marked as an error, with the `status.code` and `status.message` tags set properly.  I guess my issue was just with Zipkin.

So I'm removing the extra flags as you suggested.

Reply:
Great glad it worked :)